pgmpy/models/BayesianNetwork.py

```diff
@@ -1364,7 +1363,51 @@ def simulate(
                 model.add_cpds(new_cpd)
                 evidence[new_var] = 0
 
-        # Step 3: If no evidence do a forward sampling
+        # Step 3: If missing_prob; include missing values in samples.
+        if missing_prob:
+            if isinstance(missing_prob, list):
+                for cpd in missing_prob:
+                    if not isinstance(cpd, TabularCPD):
+                        raise ValueError(
+                            "Missing probability should only contain TabularCPD."
+                        )
+            else:
+                if isinstance(missing_prob, TabularCPD):
+                    missing_prob = [missing_prob]
+                else:
+                    raise ValueError("Missing probability should be TabularCPD.")
+
+            for cpd in missing_prob:
+                variable = cpd.variables[0]
+
+                if "*" not in variable:
+                    raise ValueError(
+                        "TabularCPD variable should end with * symbol to represent missingnness variable."
```

Comment:
I would suggest trying to add the "Got {}" part (as in the comment above). I think it makes the error message a bit more informative. For example, here we can just add something like: `Got {variable}`